features/topology-map/plugins/org.opennms.features.topology.plugins.topo.history/src/test/java/org/opennms/features/topology/plugins/topo/BundleContextHistoryManagerTest.java

```diff
@@ -201,12 +241,83 @@ public void testHistoryManager() throws IOException {
         Assert.assertEquals(historyHash5, properties.get(user1));
     }
 
+    @Test
+    public void verifySearchCriteriaPersistence() {
+        String fragment = historyManager.saveOrUpdateHistory("admin", graphContainerMock);
+        historyManager.applyHistory(fragment, graphContainerMock);
+
+        List<Criteria> resultingCriteria = capturedCriteria.getValues();
+
+        Assert.assertNotNull(resultingCriteria);
+        Assert.assertThat(availableCriteria, containsInAnyOrder(resultingCriteria.toArray()));
+    }
+
     private static Properties loadProperties() throws IOException {
         Properties props = new Properties();
         props.load(new FileInputStream(new File(DATA_FILE_NAME)));
         return props;
     }
 
+    private void initAvailableProvidersAndCriteria() {
+        CategoryProvider vertexProvider = new CategoryProvider() {
```

Comment:
This looks fine to me, but why do we have a `CategoryProviderTest` in addition to this? We should remove the test/dummy implementation. This is true for all test provideres if they are not used anywhere else.